### PR TITLE
core: Retry remote core client three times before failing out

### PR DESCRIPTION
**Pull request by renaynay**
Resolves https://github.com/celestiaorg/celestia-node/pull/289.

### CODE REVIEW

File: core/client_test.go [hunk 1]
```diff
@@ -102,3 +103,21 @@ func TestRemoteClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
...
+	// deliberately stop remote
+	err := remote.Stop()
+	require.NoError(t, err)
+
+	client, err := NewRemote(protocol, ip)
+	require.NoError(t, err)
+
+	_, err = client.Block(context.Background(), nil)
+	require.Error(t, err)
+	assert.ErrorContains(t, err, "giving up after 3 attempt(s)")
```
Comment (liamsi) [inline core/client_test.go:122]: Can't this be tested against the error type itself instead of via string? (if it is part of their public API)

**Review by liamsi — APPROVED** [review]
To nits:
 - checking against an error string
 - yet another dependency introduced for something simple (but it's OK)

File: core/client_test.go (hunk 1 above)
Comment (renaynay) [reply core/client_test.go:122]: They don't have err types

File: core/client.go [hunk 2]
```diff
@@ -14,9 +16,13 @@ type Client = client.Client
 
 // NewRemote creates a new Client that communicates with a remote Core endpoint over HTTP.
 func NewRemote(protocol, remoteAddr string) (Client, error) {
-	return http.New(
+	httpClient := retryhttp.NewClient()
+	httpClient.RetryMax = 2
```
Comment (Wondertan) [inline core/client.go:20]: Why not 3? God loves Trinity

**Review by Wondertan — APPROVED** [review]
:heart:

@jbowen93, wanna test out that reconnection is now working?

File: core/client.go (hunk 2 above)
Comment (renaynay) [reply core/client.go:20]: 3 actually means 4 tries :) bc it's 2 retries plus 1 try.

**Pull request merged**